### PR TITLE
Add grpc reflection

### PR DESCRIPTION
**Pull request by elega**
### What changes are proposed in this pull request?

Add grpc reflection to make the grpc tools such as grpcui and grpcurl send request to alluxio server easier.

### Why are the changes needed?

image

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 1]
```diff
@@ -586,6 +586,16 @@ public String toString() {
           .setScope(Scope.SERVER)
           .setIsHidden(true)
           .build();
+  public static final PropertyKey GRPC_REFLECTION_ENABLED =
+      booleanBuilder(Name.GRPC_REFLECTION_ENABLED)
+          .setDefaultValue(false)
+          .setDescription("If true, grpc reflection will be enabled on master so that "
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:592]: if it only works on master then MASTER_GRPC_DEBUG_ENABLED? Or is it possible to work on worker/job worker as well?

File: core/common/src/main/java/alluxio/grpc/GrpcServerBuilder.java [hunk 2]
```diff
@@ -260,6 +264,9 @@ public GrpcServerBuilder sslContext(SslContext sslContext) {
   public GrpcServer build() {
     addService(new GrpcService(new ServiceVersionClientServiceHandler(mServices))
         .disableAuthentication());
+    if (mGrpcReflectionEnabled) {
+      addService(new GrpcService(ProtoReflectionService.newInstance()).disableAuthentication());
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/grpc/GrpcServerBuilder.java:268]: add a comment here explain why auth is disabled

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Mostly LGTM except minor comments. Thanks for the work!

File: core/common/src/main/java/alluxio/conf/PropertyKey.java (hunk 1 above)
Comment (elega) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java:592]: it works on all grpc servers

Comment (elega) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java:592]: i'll change the name

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 3]
```diff
@@ -586,6 +586,17 @@ public String toString() {
...
+  public static final PropertyKey GRPC_REFLECTION_ENABLED =
+      booleanBuilder(Name.GRPC_REFLECTION_ENABLED)
+          .setDefaultValue(false)
+          .setDescription("If true, grpc reflection will be enabled on alluxio grpc servers, "
+              + "including masters, workers, job masters and job workers. "
+              + " This makes grpc tools such as grpcurl or grpcui can send grpc requests to "
+              + "the master server easier without knowing the protobufs. "
+              + "This is a debug option.")
+          .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
+          .setScope(Scope.MASTER)
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:598]: scope should be ALL?

**Review by jiacheliu3 — APPROVED** [review]
LGTM with one minor comment, thanks!

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**